test/python/pulse/compiler_passes/test_set_sequence.py

```diff
@@ -0,0 +1,379 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test SetSequence"""
+from test import QiskitTestCase
+from ddt import ddt, named_data, unpack
+
+from qiskit.pulse import (
+    Constant,
+    Play,
+    Delay,
+    ShiftPhase,
+)
+
+from qiskit.pulse.ir import (
+    SequenceIR,
+)
+
+from qiskit.pulse.model import QubitFrame, Qubit, MixedFrame
+from qiskit.pulse.transforms import (
+    AlignLeft,
+    AlignRight,
+    AlignSequential,
+    AlignFunc,
+    AlignEquispaced,
+)
+from qiskit.pulse.compiler import MapMixedFrame, SetSequence
+from qiskit.pulse.exceptions import PulseError
+from .utils import PulseIrTranspiler
+
+
+@ddt
+class TestSetSequenceParallelAlignment(QiskitTestCase):
+    """Test SetSequence pass with Parallel Alignment"""
+
+    ddt_named_data = [["align_left", AlignLeft()], ["align_right", AlignRight()]]
+
+    def _get_pm(self) -> PulseIrTranspiler:
+        pm = PulseIrTranspiler()
+        pm.append(MapMixedFrame())
+        pm.append(SetSequence())
+        return pm
+
+    @named_data(*ddt_named_data)
```

Comment:
I tested everything against all relevant alignment types. It's kind of an overkill, with the logic located at a parent abstract class.